### PR TITLE
[core] Support for @mui packages

### LINKED ISSUE #2527: [DataGrid] The DataGrid is not compatible with the new package name of the core repository
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When you replace the old `@material-ui/*` with the newer `@mui/*` packages for the rc versions, the datagrid is breaking.

```
Could not find dependency: '@material-ui/core' relative to '/node_modules/@mui/x-data-grid/dist/index-esm.js'
```

### Expected behavior 🤔

It should work with the new package names

### Steps to reproduce 🕹

Click link https://codesandbox.io/s/vigilant-noyce-xx89p?file=/src/Demo.tsx

or alternatively:

1. Remove any `@material-ui/*` packages
2. Add `@mui/*` packages
3. Restart app
4. 💥

### Context 🔦

No context, but a simple workaround for now. 

You can add this to package.json to alias the new packages to the old
```json
    "@material-ui/core": "npm:@mui/material@5.0.0-rc.0",
    "@material-ui/styles": "npm:@mui/styles@5.0.0-rc.0",
```

You can also do this via webpack but for some reason, it worked with storybook's webpack but not with nextjs's webpack.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
npx: installed 2 in 3.184s

  System:
    OS: macOS 11.4
  Binaries:
    Node: 14.17.3 - ~/.volta/tools/image/node/14.17.3/bin/node
    Yarn: 3.0.1 - ~/.volta/tools/image/yarn/1.22.10/bin/yarn
    npm: 6.14.13 - ~/.volta/tools/image/node/14.17.3/bin/npm
  Browsers:
    Chrome: 92.0.4515.159
    Edge: Not Found
    Firefox: Not Found
    Safari: 14.1.1
```
</details>

### Order ID 💳 (optional)

28357

### PR DESCRIPTION
**Pull request by m4theushw**
Fixes #2527

### CODE REVIEW

File: packages/grid/x-grid-data-generator/src/renderer/renderEditProgress.tsx [hunk 1]
```diff
@@ -89,7 +90,7 @@ function EditProgress(props: GridRenderEditCellParams) {
 
   const handleRef: SliderProps['ref'] = (element) => {
     if (element) {
-      element.querySelector<HTMLElement>('[role="slider"]')!.focus();
+      element.querySelector<HTMLElement>('[type="range"]')!.focus();
```
Comment (m4theushw) [inline packages/grid/x-grid-data-generator/src/renderer/renderEditProgress.tsx:93]: Fixes #2556

File: docs/scripts/formattedTSDemos.js [hunk 2]
```diff
@@ -21,13 +21,8 @@ const { fixBabelGeneratorIssues, fixLineEndings } = require('./helpers');
 
 const tsConfig = typescriptToProptypes.loadConfig(path.resolve(__dirname, '../tsconfig.json'));
 
-const unwrap = babel.createConfigItem(
-  require('../node_modules/@material-ui/monorepo/packages/babel-plugin-unwrap-createstyles'),
-);
-
 const babelConfig = {
   presets: ['@babel/preset-typescript'],
-  plugins: [unwrap],
```
Comment (m4theushw) [inline docs/scripts/formattedTSDemos.js:30]: In the core, this plugin is not used:

https://github.com/mui-org/material-ui/blob/1b563a153f5b2736f79a3f45573e2e2b66ba0a9a/docs/scripts/formattedTSDemos.js#L26

File: packages/grid/x-grid/README.md [hunk 3]
```diff
@@ -15,11 +15,12 @@ npm install @mui/x-data-grid-pro
 yarn add @mui/x-data-grid-pro
 ```
 
-This component has two peer dependencies that you will need to install as well.
+This component has three peer dependencies that you will need to install as well.
```
Comment (flaviendelangle) [inline packages/grid/x-grid/README.md:18]: This makes this PR breaking no ?

File: docs/src/pages/components/data-grid/filtering/CustomRatingOperator.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
-import { makeStyles } from '@material-ui/styles';
-import { Rating } from '@material-ui/lab';
+import { makeStyles } from '@mui/styles';
+import { Rating } from '@mui/material';
```
Comment (flaviendelangle) [inline docs/src/pages/components/data-grid/filtering/CustomRatingOperator.tsx:3]:
```suggestion
import Rating from '@mui/material/Rating';
```

File: docs/src/pages/components/data-grid/filtering/CustomRatingOperator.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import { makeStyles } from '@material-ui/styles';
-import { Rating } from '@material-ui/lab';
+import { makeStyles } from '@mui/styles';
+import { Rating } from '@mui/material';
```
Comment (flaviendelangle) [inline docs/src/pages/components/data-grid/filtering/CustomRatingOperator.js:4]:
```suggestion
import Rating from '@mui/material/Rating';
```

File: docs/src/pages/components/data-grid/filtering/ExtendNumericOperator.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import { makeStyles } from '@material-ui/styles';
-import { Rating } from '@material-ui/lab';
+import { makeStyles } from '@mui/styles';
+import { Rating } from '@mui/material';
```
Comment (flaviendelangle) [inline docs/src/pages/components/data-grid/filtering/ExtendNumericOperator.js:4]:
```suggestion
import Rating from '@mui/material/Rating';
```

File: docs/src/pages/components/data-grid/filtering/ExtendNumericOperator.tsx [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
-import { makeStyles } from '@material-ui/styles';
-import { Rating } from '@material-ui/lab';
+import { makeStyles } from '@mui/styles';
+import { Rating } from '@mui/material';
```
Comment (flaviendelangle) [inline docs/src/pages/components/data-grid/filtering/ExtendNumericOperator.tsx:3]:
```suggestion
import Rating from '@mui/material/Rating';
```

File: packages/grid/_modules_/grid/components/panel/GridPanel.test.tsx [hunk 8]
```diff
@@ -10,7 +10,7 @@ import {
   useGridApiRef,
   GridApiContext,
 } from '@mui/x-data-grid';
-import { Popper } from '@material-ui/core';
+import { Popper } from '@mui/material';
```
Comment (flaviendelangle) [inline packages/grid/_modules_/grid/components/panel/GridPanel.test.tsx:13]:
```suggestion
import Popper from '@mui/material/Popper';
```

**Comment by flaviendelangle** [comment]
image

https://deploy-preview-2571--material-ui-x.netlify.app/storybook/?path=/story/x-grid-tests-pagination--page-2-prop-snap

I can't reproduce the snapshot buggy behavior on your preview

The other snapshot diffs seems logical to me

File: packages/grid/x-grid/README.md (hunk 3 above)
Comment (m4theushw) [reply packages/grid/x-grid/README.md:18]: No, the styles package is already mentioned as a peer dependency in https://material-ui.com/components/data-grid/getting-started/#installation. This change is only syncing the README with the docs.

**Comment by m4theushw** [comment]
@flaviendelangle About the buggy snapshot. The test environment was still using the old @material-ui/core. There're only 10 differences now, which are because of the new palette.

**Review by flaviendelangle — APPROVED** [review]

**Pull request merged**